src/plugins/mission/mission_impl.cpp

```diff
@@ -289,6 +290,38 @@ MissionImpl::convert_to_int_items(const std::vector<MissionItem>& mission_items)
     _mission_data.gimbal_v2_in_control = false;
 
     for (const auto& item : mission_items) {
+        if (item.vehicle_action == VehicleAction::Takeoff) {
+            // There is a vehicle action that we need to send.
+
+            // Current is the 0th waypoint
+            uint8_t current = ((int_items.size() == 0) ? 1 : 0);
+
+            uint8_t autocontinue = 1;
+
+            const int32_t x = int32_t(std::round(item.latitude_deg * 1e7));
+            const int32_t y = int32_t(std::round(item.longitude_deg * 1e7));
+            float z = item.relative_altitude_m;
+            MAV_FRAME frame = MAV_FRAME_GLOBAL_RELATIVE_ALT_INT;
+
+            MAVLinkMissionTransfer::ItemInt next_item{
+                static_cast<uint16_t>(int_items.size()),
+                frame,
+                MAV_CMD_NAV_TAKEOFF,
+                current,
+                autocontinue,
+                NAN,
+                NAN,
+                NAN,
+                NAN,
+                x,
+                y,
+                z,
+                MAV_MISSION_TYPE_MISSION};
+
+            _mission_data.mavlink_mission_item_to_mission_item_indices.push_back(item_i);
+            int_items.push_back(next_item);
+        }
+
```

Comment:
If you do a TAKEOFF, then you don't need a WAYPOINT afterwards, I think. I'm not 100% sure how this would behave.